proposals/0000-template.md

```diff
@@ -1,188 +1,151 @@
-implemented (by you or anyone else), CLC approval will be needed for these
-changes, via a CLC proposal made by the implementor.
-By signalling those changes now, at the proposal stage, the CLC will be alerted
-and have an opportunity to offer feedback, and agreement in principle.
-
-See [GHC base libraries](https://github.com/Ericson2314/tech-proposals/blob/ghc-base-libraries/proposals/accepted/051-ghc-base-libraries.rst?rgh-link-date=2023-07-09T17%3A01%3A15Z)
-for some useful context.
-
-Therefore, in this section:
-
-* If your proposal makes any changes to the API of `base` (including its
-  exports, types, semantics, and performance), please specify these changes
-  in this section.
-
-* If your proposal makes any change to the API of `ghc-experimental`, please
-  also specify these changes.
-
-If you propose to change both, use subsections, so that the changes are clearly
-distinguished.
-Similarly, if any other libraries are affected, please lay it all out here.
+> The program may be run multiple times (e.g. with different test data), and the coverage data from the separate runs is accumulated in the .tix file.
 
-## Examples
+If there already is a `.tix` file, but this file was created for a different version of the program, then the program crashes at the time the RTS is started.
+Both behaviours are demonstrated in the example section below.
 
-This section illustrates the specification through the use of examples of the
-language change proposed. It is best to exemplify each point made in the
-specification, though perhaps one example can cover several points. Contrived
-examples are OK here. If the Motivation section describes something that is
-hard to do without this proposal, this is a good place to show how easy that
-thing is to do with the proposal.
+This behaviour is only useful for a completely manual way of interacting with instrumented executables.
+It is a nuisance for tool developers who have to manually ensure to always remove
+`.tix` files, and for users not familiar with this behaviour.
+For example, here is an example on [Stack Overflow](https://stackoverflow.com/questions/28416827/cabal-and-hpc-and-errors-when-running-tests-with-code-coverage) of a user bitten by
+the fact that the program crashes if there is a `.tix` file still around. None of the other users was able to correctly identify the underlying problem.
+And here is an issue that describes that even `cabal` was not able to correctly implement the logic for enabling code coverage in a project: [Cabal Issue #7384](https://github.com/haskell/cabal/issues/7384)
 
-## Effect and Interactions
+If a user wants to combine the results of multiple runs, then this behaviour can be trivially implemented in userland. 
+The `hpc` binary has the subcommand `hpc combine` which allows to do this on the commandline, and the HPC library  on [Hackage](https://hackage.haskell.org/package/hpc) provides the tools to do that programmatically from within Haskell programs.
```

Comment:
“trivial” is a stretch. If I want to get coverage on binary `a`, which is called indirectly by some other processes (a test suite driver, or even application logic that happens to shell out to `a`), then it is going to be rather tricky to copy away the `.tix` file in time for the next run. I believe I have relied on that behavior before, so it’s not theoretical.

It seems more trivial to delete the `.tix` file in between runs :-)

Maybe the old behaviour could be guarded by a RTS flag?

Reply:
> “trivial” is a stretch. If I want to get coverage on binary a, which is called indirectly by some other processes (a test suite driver, or even application logic that happens to shell out to a), then it is going to be rather tricky to copy away the .tix file in time for the next run. I believe I have relied on that behavior before, so it’s not theoretical.

Thanks, I removed the word "trivially", and turned nobody to "very few people"  ;) The usecase you describe is actually one that I have too, but where I ran into the following restriction of cabal: https://github.com/haskell/cabal/issues/9251

> Maybe the old behaviour could be guarded by a RTS flag?
I have added this in the "Alternatives" section. I could also implement that RTS flag, if the comittee decides that this alternative is better.